esp-hal-common/src/clocks_ll/esp32.rs

```diff
@@ -0,0 +1,381 @@
+                    dcur = 0;
+                    bw = 1;
+                }
+
+                XtalFrequency::RtcXtalFreqOther(_) => {
+                    div_ref = 12;
+                    div7_0 = 224;
+                    div10_8 = 4;
+                    lref = 0;
+                    dcur = 0;
+                    bw = 0;
+                }
+            }
+
+            i2c_writereg_rtc(
+                I2C_BBPLL,
+                I2C_BBPLL_HOSTID,
+                I2C_BBPLL_ENDIV5,
+                BBPLL_ENDIV5_VAL_480M,
+            );
+
+            i2c_writereg_rtc(
+                I2C_BBPLL,
+                I2C_BBPLL_HOSTID,
+                I2C_BBPLL_BBADC_DSMP,
+                BBPLL_BBADC_DSMP_VAL_480M,
+            );
+        }
+
+        i2c_bbpll_lref = (lref << 7) | (div10_8 << 4) | (div_ref);
+        i2c_bbpll_div_7_0 = div7_0;
+        i2c_bbpll_dcur = (bw << 6) | dcur;
+        i2c_writereg_rtc(
+            I2C_BBPLL,
+            I2C_BBPLL_HOSTID,
+            I2C_BBPLL_OC_LREF,
+            i2c_bbpll_lref,
+        );
+
+        i2c_writereg_rtc(
+            I2C_BBPLL,
+            I2C_BBPLL_HOSTID,
+            I2C_BBPLL_OC_DIV_7_0,
+            i2c_bbpll_div_7_0,
+        );
+
+        i2c_writereg_rtc(
+            I2C_BBPLL,
+            I2C_BBPLL_HOSTID,
+            I2C_BBPLL_OC_DCUR,
+            i2c_bbpll_dcur,
+        );
+    }
+}
+
+#[ram]
+pub(crate) fn esp32_rtc_bbpll_enable() {
+    let rtc_cntl = unsafe { &*crate::pac::RTC_CNTL::ptr() };
+
+    unsafe {
+        rtc_cntl.options0.modify(|_, w| {
+            w.bias_i2c_force_pd()
+                .clear_bit()
+                .bb_i2c_force_pd()
+                .clear_bit()
+                .bbpll_force_pd()
+                .clear_bit()
+                .bbpll_i2c_force_pd()
+                .clear_bit()
+        });
+
+        // reset BBPLL configuration
+        i2c_writereg_rtc(
+            I2C_BBPLL,
+            I2C_BBPLL_HOSTID,
+            I2C_BBPLL_IR_CAL_DELAY,
+            BBPLL_IR_CAL_DELAY_VAL,
+        );
+        i2c_writereg_rtc(
+            I2C_BBPLL,
+            I2C_BBPLL_HOSTID,
+            I2C_BBPLL_IR_CAL_EXT_CAP,
+            BBPLL_IR_CAL_EXT_CAP_VAL,
+        );
+        i2c_writereg_rtc(
+            I2C_BBPLL,
+            I2C_BBPLL_HOSTID,
+            I2C_BBPLL_OC_ENB_FCAL,
+            BBPLL_OC_ENB_FCAL_VAL,
+        );
+        i2c_writereg_rtc(
+            I2C_BBPLL,
+            I2C_BBPLL_HOSTID,
+            I2C_BBPLL_OC_ENB_VCON,
+            BBPLL_OC_ENB_VCON_VAL,
+        );
+        i2c_writereg_rtc(
+            I2C_BBPLL,
+            I2C_BBPLL_HOSTID,
+            I2C_BBPLL_BBADC_CAL_7_0,
+            BBPLL_BBADC_CAL_7_0_VAL,
+        );
+    }
+}
+
+#[inline(always)]
+unsafe fn i2c_writereg_rtc(block: u32, block_hostid: u32, reg_add: u32, indata: u32) {
+    const ROM_I2C_WRITEREG: u32 = 0x400041a4;
+
+    // cast to usize is just needed because of the way we run clippy in CI
```

Comment:
I think it was actually a hard compile error - will check that

Reply:
checked that: it's a hard compile error since in CI we run clippy for the host target (because we can't run clippy with the esp toolchain) ... so we need it like this unfortunately